clang/lib/CodeGen/CGDebugInfo.h

```diff
@@ -346,6 +348,15 @@ class CGDebugInfo {
       const FieldDecl *BitFieldDecl, const llvm::DIDerivedType *BitFieldDI,
       llvm::ArrayRef<llvm::Metadata *> PreviousFieldsDI, const RecordDecl *RD);
 
+  // A cache that maps artificial inlined function names used for
+  // __builtin_verbose_trap to subprograms.
+  llvm::StringMap<llvm::DISubprogram *> InlinedTrapFuncMap;
```

Comment:
Just to check. What's the life time of CGDebugInfo? Is it a single instance per complication unit? If it's longer than that (i.e. the same instance gets used for multiple compilation units) then we might cache the sub programs for too long.

Reply:
It doesn't live longer than a compilation unit. `CodeGenModule::DebugInfo` is created in the constructor of `CodeGenModule`.